proposals/bip-001.md

```diff
@@ -0,0 +1,107 @@
+driver implementor finds that there's some feature missing to complete an AST to
+UAST conversion, this new feature is requested to be added through a Pull
+Request. While this process is quick, it doesn't scale well: a single language
+need is not the best background for a universal feature request discussion, and
+it easies a process that may degenerate in a UAST that becomes the sum of each
+single language feature, which is something we should avoid. That's why
+introducing a more formal process, requiring a larger background and use cases,
+should help visualising which features are general enough, which ones are just
+special cases of a more general ones, which ones are just details that may be
+left for a future version, etc.
+
+## Workflow
+
+A BIP process should begin with an author proposing an idea. This idea should be
+as focused as possible.
+
+The author must:
+- Discuss the idea publicly, gather early feedback and make sure that the idea
+  is not guaranteed to be rejected.
+- Present a draft. It should take the next available BIP number and its status
+  should be **Draft**
+- Reach consensus. Feedback should be added back to the proposal.
+
+Once the BIP has been completed and reviewed it may be **Accepted** or
+**Rejected** (it may have not been a good idea after all). In any case it's
+important to have a record of the discussion so the proposal will be archived.
+
+
+## Specification
+
+Each BIP should have the following parts
+
+1. Title
+
+2. Header with metadata
+
+3. Abstract -- a short description of the issue being addressed
+
+4. Rationale -- proposal motivation. It should be clearly explain why current
+   design is inadequate to address the problem that this BIP solves, and the use
+   cases for it.
+
+5. Specification -- The technical specification should clearly describe the
+   change details. It should be detailed enough to allow language driver
+   implementors understand whether and how they should make use of this new
+   feature or change.
+
+6. Alternatives -- The alternatives should describe how the issue is currently
+   solved, how it could be solved if this change is not accepted, or what
+   alternative designs could be implemented.
+
+7. Impact -- Which languages are affected, how frequent this feature appears in
+   each language, which existing drivers are affected and how they should be
+   modified must be documented here.
+
+8. References -- Links to language specs describing related features or other
+   kind of related information.
+
+### Format and Templates
+
+BIPs must be written in Markdown format. [BIP 0](bip-000.md) should be used as a
+template.
+
+### Header
+
+Each BIP must begin with a header that includes the following information:
+
+- BIP number
+- Title
+- Author. List of authors with real names.
+- Status. One of Draft | Accepted | Rejected
+- Created. Date of creation
+- Updated. Date of last update
+
```

Comment:
The proposals should be timeless IMHO. A BIP may include information about what version of the UAST includes the implementation though.